### ISSUE TITLE
Weird HDMI output on the fml13v01 board

### ISSUE THREAD

**Issue by SpieringsAE** [DC-DeepComputing/DC-linux#1]
After some tweaking on the config I managed to get arch riscv booting with this kernel.

But the hdmi output is weird, the image does end up forming, but sometimes it takes quite a bit of time.

The new output appears as noise at first and then slowly resolves to the actual output.

Is that a known issue with this kernel?

branch/fork with my changes: https://github.com/SpieringsAE/DC-linux/tree/wip/6.13-rc4

Looking forward to seeing this kernel getting fleshed out more! Thank you for the great board!

**Comment by sandiecao-dc** [comment]
DC-linux kernel doesn't have this issue. 
If you can't fix it, you may try to update uboot from here: https://github.com/DC-DeepComputing/fml13v01/releases/tag/V1.0.

**Comment by SpieringsAE** [comment]
I will try again with the newest changes some time soon and then try to update u-boot.

I didn't notice it as strong with the ubuntu/fedora images (still a little bit fuzzy during fast movement like the initial kernel/init console output), so my assumption was that it was this version of the hdmi driver that was at fault.

**Comment by sandiecao-dc** [comment]
Please note DC-linux kernel doesn't support GPU,  fml13v01 ubuntu/fedora images supports GPU. so DC-linux kernel can't work as smooth as  fml13v01 ubuntu/fedora images kernel.

**Comment by SpieringsAE** [comment]
I am aware but thank you for the info anyway, I did also try the pwm driver on the mailing list, but it seems to invert the backlight logic. I saw the fix for it in fml13v01-linux, but I don't think that is going to fly upstream. Still thinking of how to fix that. Maybe an extra bool property for the pwm-backlight driver for inverted pwm.

**Comment by sandiecao-dc** [comment]
The EC of fml13v01 board handles pwm-backlight. And the user space uses cros-ec to communicate with EC.  patches of this part should be upload later.

**Comment by SpieringsAE** [comment]
it seems to come from the SOC pwm generator:
https://github.com/DC-DeepComputing/fml13v01-linux/blob/e711a7467842cbc9d5cc1e40ec943c5f6537336c/arch/riscv/boot/dts/starfive/jh7110-deepcomputing-fml13v01.dts#L82-L120
https://github.com/DC-DeepComputing/fml13v01-linux/blob/e711a7467842cbc9d5cc1e40ec943c5f6537336c/arch/riscv/boot/dts/starfive/jh7110-deepcomputing-fml13v01.dts#L434-L438

I have also controlled the backlight with this pwm, unless somehow the EC is in between this pwm and the actual backlight pwm, or is hooked up in parallel but that seems highly unlikely to me.

**Comment by sandiecao-dc** [comment]
Yes，the EC controls the power of backlight.  above patches control the brightness of backlight.

**Comment by SpieringsAE** [comment]
> DC-linux kernel doesn't have this issue. If you can't fix it, you may try to update uboot from here: https://github.com/DC-DeepComputing/fml13v01/releases/tag/V1.0.

So I have updated u-boot to the newest version from the V1.1 release, the dark startup screen and the fix for the slow kernel load are very nice.  
But this issue still persists, I have also updated to a more recent set of patches from this repo but that didn't change anything either.

The hdmi driver in the fml13v01-linux tree seems to work better in some way. I haven't seen an updated patch from Keith Zhao in quite a while, do you know if this patch set is still being worked on by the people at starfive? This tree also seems to be using the v4 version of that patch series but there is a newer v5 on the mailing list, but it seems to need a rebase. I will try to do that this weekend on my tree.

**Comment by sandiecao-dc** [comment]
Yes, we got the hdmi driver from starfive v4 patch series. And we only update to newer version when the newer starfive patch is accepted by Linux.
